### PR TITLE
Tweak WP8 benchmark

### PR DESCRIPTION
**Pull request by jorisdral**

### CODE REVIEW

File: bench/macro/lsm-tree-bench-wp8.hs [hunk 1]
```diff
@@ -394,13 +396,14 @@ doRun gopts opts = do
               | otherwise      = sequentialIterations
         time <- timed_ $
           benchmarkIterations
-            check
+            (\b y -> check b y >> when (b `mod` (opts.batchCount `div` 100) == 0) (putChar '.'))
```
Comment (dcoutts) [inline bench/macro/lsm-tree-bench-wp8.hs:399]: Pull the progress calc out so we don't repeat it every time.

I'm not sure this is the best place to put it. I've also got a tidy-up branch where I separate the `check` into a `shouldOutput :: Bool` and `doOutput` action, to really minimise the effect when we're not doing testing. But progress reporting should be unconditional.

File: bench/macro/lsm-tree-bench-wp8.hs [hunk 2]
```diff
@@ -548,6 +551,11 @@ pipelinedIterations :: (Int -> LookupResults -> IO ())
                     -> LSM.TableHandle IO K V B
                     -> IO ()
 pipelinedIterations output !initialSize !batchSize !batchCount !seed tbl_0 = do
+    n <- getNumCapabilities
+    unless (n == 2) $
+      printf "WARNING: the pipelined benchmark is running with %d capabilities \
+             \but it is recommended to run with 2. Pass the -N2 RTS option.\n" n
```
Comment (dcoutts) [inline bench/macro/lsm-tree-bench-wp8.hs:557]: Well, yes and no. It's also interesting to see what performance it has if one runs both threads on one core. There's still concurrency available there.

File: bench/macro/lsm-tree-bench-wp8.hs (hunk 1 above)
Comment (jorisdral) [reply bench/macro/lsm-tree-bench-wp8.hs:399]: I'm not sure I understand what you would like this to look like. Wouldn't we need to compute the `mod` to know if we've made progress? Or should we print a dot by default?

Comment (dcoutts) [reply bench/macro/lsm-tree-bench-wp8.hs:399]: The `mod` is fine. Just pull out and name the `progress = opts.batchCount `div` 80`. It's clearer that way.

File: bench/macro/lsm-tree-bench-wp8.hs (hunk 2 above)
Comment (jorisdral) [reply bench/macro/lsm-tree-bench-wp8.hs:557]: I've changed to an info message that just tells the user how many threads are in use

**Review by dcoutts — APPROVED** [review]

**Pull request merged**